spec/zero_downtime_migrations/validation/add_foreign_key_spec.rb

```diff
@@ -0,0 +1,77 @@
+RSpec.describe ZeroDowntimeMigrations::Validation::AddForeignKey do
+  let(:error) { ZeroDowntimeMigrations::UnsafeMigrationError }
+
+  context "when the database supports_validate_constraints" do
+    context "adding a foreign key and not specifying validate" do
+      let(:migration) do
+        Class.new(ActiveRecord::Migration[5.0]) do
+          def change
+            create_table :order
+            add_column :users, :order_id, :integer
+            add_foreign_key :users, :order
+          end
+        end
+      end
+
+      it "raises an unsafe migration error" do
```

Comment:
I think you could put this in a `shared_example` WDYT? 🤔 

```rb
RSpec.shared_examples 'returns exception error' do
  let(:error) { ZeroDowntimeMigrations::UnsafeMigrationError }

  it 'raises an unsafe migration error' do
    expect { migration.migrate(:up) }.to raise_error(error)
  end
end

RSpec.describe ZeroDowntimeMigrations::Validation::AddForeignKey do
  context 'when the database supports_validate_constraints' do
    context 'adding a foreign key and not specifying validate' do
      let(:migration) do
        ...
      end

      include_examples 'returns exception error'
    end
  end
end
```
source: https://relishapp.com/rspec/rspec-core/docs/example-groups/shared-examples